### ISSUE TITLE
Potential Issue on Collection

### ISSUE THREAD

**Issue by GeorgeGo** [ebullient/obsidian-task-collector#132]
I'm not sure if this is or isn't intended behavior and I wasn't sure of the best method to get feedback.

When performing Collect Tasks with the Remove Checkbox option enabled, items that are subtasks don't have their checkboxes removed.

For example,

- [x] A
    - [x] B
    - [x] C 

Becomes,

## Log
- A
    - [x] B
    - [x] C

I'm using version 1.0.5 of Task Collector with a fresh vault using the default settings enabling Remove Checkboxes.

**Comment by ebullient** [comment]
This is a perfect method. Thanks for raising the issue! I can add some tests and fix this to ensure that happens.. oversight on my part. ;)

**Comment by GeorgeGo** [comment]
I'm fine writing the fix for it. I just wanted to confirm that it was an issue first.

**Comment by ebullient** [comment]
yes! that would be super helpful. You know the drill with tests. ;)

**Comment by GeorgeGo** [comment]
I'm including this as a note for myself but I also noticed an issue where incomplete tasks with complete subtasks would get incorrectly collected.

**Comment by ebullient** [comment]
I have some tests for incomplete parents w/ completed kids. I'm surprised that one is happening.. 
Relationship/collection behavior for tasks w/ subtasks is... interesting.

**Comment by GeorgeGo** [comment]
I’ll have to double check the behavior and make sure that it’s consistent/not due to something else. 

I’ll update the test cases once I’ve better defined what all is going on.

I’m still planning on addressing the main collect tasks issue this issue addresses, just been bogged down with work.

**Comment by GeorgeGo** [comment]
Also is it intended for the 'Task Mark' operation to overwrite the time information on a task that already has that specific mark?

**Comment by ebullient** [comment]
yes. Every flavor of mark has an undo that is applied before the next mark is added...

**Comment by ebullient** [comment]
> Also is it intended for the 'Task Mark' operation to overwrite the time information on a task that already has **that specific mark**?

I missed that nuance. We could probably check to see if the mark is the same, and skip it if so.
